backend/cmm_helpers.ml

```diff
@@ -3313,6 +3319,41 @@ let prefetch_offset ~is_write locality (arg1, arg2) dbg =
 let ext_pointer_prefetch ~is_write locality arg dbg =
   prefetch ~is_write locality (int_as_pointer arg dbg) dbg
 
+let native_pointer_cas size (dst, compare_with, set_to) dbg =
+  let op = Catomic { op = CAS; size } in
+  if_operation_supported op ~f:(fun () ->
+      Cop (op, [compare_with; set_to; dst], dbg))
+
+let ext_pointer_cas size (dst, compare_with, set_to) dbg =
+  let op = Catomic { op = CAS; size } in
+  if_operation_supported op ~f:(fun () ->
+      Cop (op, [compare_with; set_to; int_as_pointer dst dbg], dbg))
```

Comment:
Is this instruction setting the pointer itself or the memory block pointed by it? If latter, I think we need a load here (similarly to fetch_and_add), and I'm not sure how the tests pass.

Reply:
Hmm, not sure what you mean - it's setting the memory pointed to by the ext pointer. It works in the same manner as the native_pointer case - it just needs to decode the ext pointer first (int_as_pointer).

Reply:
ah, yes, it is fine, selectgen does the right thing here.